spec/PeriodicBackgroundSync-index.bs

```diff
@@ -59,23 +70,222 @@ Introduction {#intro}
+
+Resource Usage {#resources}
+============================
+
+<em>This section is non-normative.</em>
+
+A website will most likely download resources from the network when processing a [=periodicsync event=]. The underlying operating system may launch the user agent to dispatch these events, and may keep it awake for a pre-defined duration to allow processing of the events. Both cause battery drain.
+The user agent should cap the duration and frequency of these events to limit resource usage by websites when the user has navigated away.
+
+Large resources should be downloaded by registering a [=background fetch=] via the {{BackgroundFetchManager}} interface.
+
+In addition, the user agent should consider other factors such as user engagement with the [=origin=], and any user indications to temporarily reduce data consumption, such as a Data Saving Mode, to adjust the frequency of [=periodicsync event|periodicSync events=].
+
+Algorithms {#algorithms}
+=========================
+## <dfn>Calculate a time to fire</dfn> ## {#caculate-time-to-fire}
+This section describes how a user agent can calculate the [=periodicsync registration/time to fire=] for a [=periodicsync registration=], |registration|. The time interval between [=periodicSync event|periodicSync events=] for |registration| MUST be greater than or equal to |registration|'s [=options/minInterval=] value.
+
+The user agent MAY include factors such as user engagement with the origin to decide this time interval, allowing origins with high user engagement to update their web apps more often. The user agent SHOULD also ensure this time interval conforms to the caps asserted by [=minimum interval for any origin=] and the [=minimum interval across origins=].
+
+The user agent MAY also decide to [=retry=] each failed [=periodicSync event=] , with the count of retries capped to [=maximum number of retries=].
+
+A possible algorithm to calculate the [=periodicsync registration/time to fire=], |timeToFire| for |registration| would involve running these steps:
+1. If [=periodicsync registration/count of retries=] for |registration| is 0, <dfn>calculate the time to fire of the first attempt of</dfn> the [=periodicSync event=] :
+    1. Let |now|, a timestmap, represent the current time.
+    1. Let |origin| represent [=periodicsync registration/service worker registration=]'s [=origin=]. Let |siteEngagementLevel|, an enumeration, represent the level of engagement the user has with |origin|. Calculate |siteEngagementPenalty|, a number, which is a factor of |siteEngagementLevel|, and is inversely proportional to it. Define |effectiveGap|, as |siteEngagementPenalty|*[=minimum interval for any origin=].
```

Comment:
I understand that this is illustrative, but I'm wondering if mentioning SES is too abstract, particularly because the units and possible ranges aren't being mentioned. WDYT?

Reply:
I should probably mention ranges. Background Fetch spec uses "number" as a type/unit so there's precedent for that. In a follow-up PR I'll add a section to describe site engagement levels, where I can specify example ranges to add specificity here, and will link to that section from here. WDYT?

Reply:
We have no interest in standardizing site engagement levels, so I think we shouldn't do that. Something much higher level like the following would work:

> Given a UA-defined `minimum interval for any origin` of 12 hours, multiplied by 3 to account for infrequent user engagement for the `origin`, set `effectiveGap` to 36 hours.

I find it a bit confusing with either wording how `effectiveGap` _isn't_ the effective gap, but rather a variable that feeds in to the `delayForOrigin`.

Reply:
I've rephrased according to your suggestion, and have renamed effectiveGap to minIntervalForOrigin.